auacm/app/modules/submission_manager/judge.py

```diff
@@ -0,0 +1,225 @@
+        submission.emit_status("compile", -1)
+        return COMPILATION_ERROR
+
+
+def execute_submission(submission, uploaded_file):
+    '''Run the submission.
+
+    TODO(djshuckerow): This method magically got out-of-hand.  Refactor.
+    This method:
+        1. detects all the input files associated with this problem.
+        2. runs the submission with each input file
+        3. checks the performance of the submission for errors (TLE, RTE, etc)
+        4. compares the output against golden test output to verify correctness
+    '''
+    # Initial setup
+    problem = submission.get_problem()
+    problem_directory = directory_for_problem(submission)
+    submission_directory = directory_for_submission(submission)
+    filename = uploaded_file.filename
+    name, ext = filename.rsplit(".", 1)
+    input_path = os.path.join(problem_directory, "in")
+    output_path = os.path.join(problem_directory, "out")
+    # Iterate over all the input files.
+    for fname in os.listdir(input_path):
+        f = os.path.join(input_path, fname)
+        if os.path.isfile(f):
+            # Prepare to run the test file.
+            test_number = int(fname.split(".")[0].strip("in"))
+            out_file = "out{0}.txt".format(test_number)
+            submission.emit_status("running", test_number)
+            max_runtime = problem.time_limit * TIMEOUT_MULTIPLIER[ext]
+            execution = JudgementThread(
+                submission, uploaded_file, f, out_file, max_runtime)
+            start_time = time.time()
+            execution.start()
+            execution.join(max_runtime)
+            # Check the execution for timeouts and runtime errors.
+            if time.time() >= start_time + max_runtime:
+                execution.process.kill()
+                submission.update_status("timeout")
+                submission.emit_status("timeout", test_number)
+                return TIMELIMIT_EXCEEDED
+            elif execution.process.poll() != 0:
+                submission.update_status("runtime")
+                submission.emit_status("runtime", test_number)
+                return RUNTIME_ERROR
+            result_path = os.path.join(submission_directory, "out")
+            # The execution is completed.  Check its correctness.
+            with open(os.path.join(output_path, out_file)) as golden_result, \
+                 open(os.path.join(result_path, out_file)) as submission_result:
+                golden_lines = golden_result.readlines()
+                submission_lines = submission_result.readlines()
+                if len(submission_lines) != len(golden_lines):
+                    submission.update_status("wrong")
+                    submission.emit_status("incorrect", test_number)
+                    return WRONG_ANSWER
+                # Use itertools.izip instead of zip to save memory.
+                for gl, sl in itertools.izip(golden_lines, submission_lines):
+                    if gl.rstrip() != sl.rstrip():
```

Comment:
Could this strip trailing whitespace in a submission? ICPC considers trailing whitespace to be incorrect, even if the the stripped versions are equal.